src/main/scala/ch/epfl/bluebrain/nexus/kg/search/QueryResultEncoder.scala

```diff
@@ -0,0 +1,48 @@
+package ch.epfl.bluebrain.nexus.kg.search
+
+import cats.syntax.show._
+import ch.epfl.bluebrain.nexus.commons.types.search.QueryResults.{ScoredQueryResults, UnscoredQueryResults}
+import ch.epfl.bluebrain.nexus.kg.config.Contexts
+import ch.epfl.bluebrain.nexus.rdf.Iri.AbsoluteIri
+import io.circe.{Encoder, Json}
+
+object QueryResultEncoder {
+
+  /**
+    * Encoder for scored query results
+    */
+  implicit val scoredEncoder: Encoder[ScoredQueryResults[AbsoluteIri]] = Encoder.instance { results =>
+    Json.obj(
+      "@context" -> Json.fromString(Contexts.searchCtxUri.show),
+      "total"    -> Json.fromLong(results.total),
+      "maxScore" -> Json.fromFloatOrString(results.maxScore),
+      "results" -> Json.arr(
+        results.results.map { res =>
+          Json.obj(
+            "resultId" -> Json.fromString(res.source.show)
+          )
+
+        }: _*
+      )
+    )
+  }
+
+  /**
+    * Encoder for unscored query results
+    */
+  implicit val unscoredEncoder: Encoder[UnscoredQueryResults[AbsoluteIri]] = Encoder.instance { results =>
+    Json.obj(
```

Comment:
you can just create the Json without the `@context` and then use `.addContext(Contexts.searchCtxUri)`

Reply:
Here you built a Json, while the preferred way is to build a graph and output it in Json format (as we did when outputing `ResourceF` or `ResourceV`. We ensure like this json-LD validity